astropy/nddata/tests/test_ccddata.py

```diff
@@ -1103,15 +1103,15 @@ def test_sliced_ccdata_to_hdu():
 def test_read_write_tilde_paths(home_is_tmpdir):
     # Test for reading and writing to tilde-prefixed paths without errors
     ccd_data = create_ccd_data()
-    filename = os.path.join("~", "test.fits")
+    filename = Path("~/test.fits")
     ccd_data.write(filename)
 
     ccd_disk = CCDData.read(filename, unit=ccd_data.unit)
     np.testing.assert_array_equal(ccd_data.data, ccd_disk.data)
 
     # Ensure the unexpanded path doesn't exist (e.g. no directory whose name is
     # a literal ~ was created)
-    assert not os.path.exists(filename)
+    assert not Path(filename).exists()
```

Comment:
This test is checking that the `CCDData` I/O-functions perform [tilde expansion like a POSIX shell](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_01). `Path` does not do tilde expansion automatically and instead expects the user to call the [`Path.expanduser()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.expanduser) method explicitly.

It is true that the code should not be converting `filename` to a `Path` because it already is one.